### PR TITLE
Changes needed for a clean exit on the FW side

### PR DESCRIPTION
**Pull request by moratom**
Also added in thread naming on Linux for easier profiling.

### CODE REVIEW

File: src/pipeline/ThreadedNode.cpp [hunk 1]
```diff
@@ -8,12 +14,19 @@ void ThreadedNode::start() {
...
+            running = false;
+        } catch(const std::runtime_error& ex){
             logger->error("Node threw exception, stopping the node. Exception message: {}", ex.what());
             running = false;
         }
     });
+#ifdef __linux__
+    auto handle = thread.native_handle();
+    pthread_setname_np(handle, fmt::format("{}-{}", getName(), id).c_str());
+#endif
```
Comment (themarpe) [inline src/pipeline/ThreadedNode.cpp:29]: Lets move this to platform specific utils
And naming format is `{}({})` -> name, id

File: src/pipeline/ThreadedNode.cpp [hunk 2]
```diff
@@ -1,5 +1,11 @@
 #include "depthai/pipeline/ThreadedNode.hpp"
 
+#include <spdlog/spdlog.h>
```
Comment (themarpe) [inline src/pipeline/ThreadedNode.cpp:3]: Needed?

File: src/pipeline/ThreadedNode.cpp [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+
+#ifdef __linux__
+    #include <pthread.h>
+#endif
```
Comment (themarpe) [inline src/pipeline/ThreadedNode.cpp:7]: Move to platform specific utils as well

**Review by themarpe — CHANGES_REQUESTED** [review]

File: src/pipeline/ThreadedNode.cpp (hunk 2 above)
Comment (moratom) [reply src/pipeline/ThreadedNode.cpp:3]: No, removed.

**Pull request merged**